### PR TITLE
Add React UI with our components

### PR DESCRIPTION
**Pull request by kevalnagda**
Fix #9

### CODE REVIEW

File: portal/client/src/libs/backendChecks.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+// TODO: should be provided from backend
+import _ from 'lodash'
+
+import constants from './backendStringConstants'
+
+const {
```
Comment (chsasank) [inline portal/client/src/libs/backendChecks.js:6]: remove most of these irrelevant. please remove

File: portal/client/.eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    'Image': true,
+  },
+  'parser': 'babel-eslint',
+  'settings': {
+    'import/resolver': {
+      'alias': {
+        'map': [
+          ['@', path.resolve(__dirname, 'src')],
+          ['react', path.resolve(__dirname, './node_modules/react')],
+          ['states', path.resolve(__dirname, './src/states')],
```
Comment (abhijeet-qure) [inline portal/client/.eslintrc.js:44]: are you using these aliases?

File: portal/client/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
```
Comment (abhijeet-qure) [inline portal/client/README.md:1]: @kevalnagda @chsasank 
Important for the OSS project to have proper README to get started on the project, lets remove this default from CRA and use our own proper README

File: portal/client/debug.log [hunk 4]
```diff
@@ -0,0 +1 @@
+[0118/111306.894:ERROR:directory_reader_win.cc(43)] FindFirstFile: The system cannot find the path specified. (0x3)
```
Comment (abhijeet-qure) [inline portal/client/debug.log:1]: lets remove this file and add the line in gitignore to not to include these `*.log` files

File: portal/client/package.json [hunk 5]
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "client",
```
Comment (abhijeet-qure) [inline portal/client/package.json:2]: Proper name please

File: portal/client/package.json [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    "watch": "^1.0.2",
+    "web-vitals": "^1.1.0",
+    "yup": "^0.29.3"
+  },
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject",
+    "lint": "npm run lint:js; npm run lint:css",
```
Comment (abhijeet-qure) [inline portal/client/package.json:40]: Can you add the TODO for us to include the Linter and prettier in git hooks?

File: portal/client/public/manifest.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
```
Comment (abhijeet-qure) [inline portal/client/public/manifest.json:2]: Lets update this file with proper names and use the icons from `src/public/icon`, look in the `qTrack` project for same, if did not copied earlier

File: portal/client/public/qTrackLogo.svg [hunk 8]
```diff
@@ -0,0 +1,136 @@
+<?xml version="1.0" standalone="no"?>
```
Comment (abhijeet-qure) [inline portal/client/public/qTrackLogo.svg:1]: Do you need this **qTrack** Logo in this qStudy project?

File: portal/client/src/App.css [hunk 9]
```diff
@@ -0,0 +1,38 @@
+.App {
```
Comment (abhijeet-qure) [inline portal/client/src/App.css:1]: Do you need any of CSS from this file? it was auto generated by CRA

File: portal/client/src/App.test.js [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import { render, screen } from '@testing-library/react';
```
Comment (abhijeet-qure) [inline portal/client/src/App.test.js:1]: lets remove this

File: portal/client/src/index.css [hunk 11]
```diff
@@ -0,0 +1,13 @@
+body {
+  margin: 0;
+  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
```
Comment (abhijeet-qure) [inline portal/client/src/index.css:3]: Can you use the IBM plex font? check the qTrack or container repo scss file for configuration of same

File: portal/client/src/libs/backendStringConstants.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
+const TB = 'tb'
```
Comment (abhijeet-qure) [inline portal/client/src/libs/backendStringConstants.js:1]: please remove unused ones

File: portal/client/src/libs/dates.js [hunk 13]
```diff
@@ -0,0 +1,6 @@
+function getFormattedDate(dateString) {
```
Comment (abhijeet-qure) [inline portal/client/src/libs/dates.js:1]: is this needed?

File: portal/client/src/libs/getFeatureToggles.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
+import _ from 'lodash'
+
+const getFeatureToggles = (userProfile) => {
```
Comment (abhijeet-qure) [inline portal/client/src/libs/getFeatureToggles.js:3]: remove this file

File: portal/client/src/libs/servicesUtils.js [hunk 15]
```diff
@@ -0,0 +1,9 @@
+import serviceConnect from './axios-connect'
+
+// simple wrapper around axios connect for simple fetch requests
+// you will need to handle the errors from catch
+function getService(url, params = {}) {
```
Comment (abhijeet-qure) [inline portal/client/src/libs/servicesUtils.js:5]: are you using this?

File: portal/client/src/pages/StudyList/index.js [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+                    <Link to="/login" className="button mt-2 is-block">                        
+                        <Icon iConClass="fa-sign-out-alt" />
+                        <span>{'Logout'}</span>
+                    </Link>
+                </div>
+            </Level.Right>
+            </Level>
+        </section>
+        <div className="is-size-5 has-text-black-bis mb-5 ml-4">
+            {'Hello, User'}
```
Comment (abhijeet-qure) [inline portal/client/src/pages/StudyList/index.js:49]: Can you show the name? if thats not possible, this is fine as well but do remove the curly brances, they are not needed here

File: portal/client/src/index.css (hunk 11 above)
Comment (abhijeet-qure) [reply portal/client/src/index.css:3]: i see that you are using it in `portal/client/src/scss/global.scss`, lets remove this file

File: portal/client/src/scss/stories.scss [hunk 17]
```diff
@@ -0,0 +1,9 @@
+.table {
```
Comment (abhijeet-qure) [inline portal/client/src/scss/stories.scss:1]: is this used in the actual project or part of component? 
naming seems to be not proper

File: portal/client/src/scss/popover.scss [hunk 18]
```diff
@@ -0,0 +1,4 @@
+.popover {
```
Comment (abhijeet-qure) [inline portal/client/src/scss/popover.scss:1]: i dont see component for this, lets remove it

File: portal/client/src/pages/AppLogin/index.js [hunk 19]
```diff
@@ -0,0 +1,27 @@
+import React, { useState } from 'react'
+import { Login } from '../../components'
+
+const AppLogin = () => {
+  const [isSubmitting, setIsSubmitting] =  useState(false)
+  const [errorMessage, setErrorMessage] = useState(null)
+  
+  const onSubmit = ()=>{
```
Comment (abhijeet-qure) [inline portal/client/src/pages/AppLogin/index.js:8]: I dont see the API call for the login, did we miss it?

File: portal/client/src/pages/StudyList/index.js [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+    const [isLoading, setLoading] = useState(false)
+    const [isSubmitting, setIsSubmitting] = useState(false)
+    const data = []
+    const columns = []
+
+    return (
+        <PageContainer
+            hasNoHeader
+            isLoading={isLoading}
+            title={'View study list'}
```
Comment (abhijeet-qure) [inline portal/client/src/pages/StudyList/index.js:18]: You dont need the `{}` unless you have some variable or express in it. I see in many places you have used the `{}` just for the string, lets remove it, as this adds the overhead to JS runtime to evaulate string uncessarily

File: portal/client/src/pages/StudyList/index.js [hunk 21]
```diff
@@ -0,0 +1,115 @@
...
+                    </Link>
+                </div>
+            </Level.Right>
+            </Level>
+        </section>
+        <div className="is-size-5 has-text-black-bis mb-5 ml-4">
+            {'Hello, User'}
+        </div>
+        <div className="ml-4 mr-4">
+            <Columns>
```
Comment (abhijeet-qure) [inline portal/client/src/pages/StudyList/index.js:52]:
```suggestion
            <Columns className="ml-4 mr-4">
```

Lets add the `className` props to the `Columns` component and use the `className` for `div` at line `51` in here

File: portal/client/src/scss/vendor.scss [hunk 22]
```diff
@@ -0,0 +1,10 @@
+@import './variables';
+@import "~@fortawesome/fontawesome-free/css/all.css";
+@import "~@fortawesome/fontawesome-free/css/fontawesome.css";
+
+@import '~bulma/bulma.sass';
+@import "~bulma-extensions/bulma-checkradio/src/sass/index.sass";
+@import '~bulma-extensions/bulma-calendar/src/sass/index.sass';
+@import '~bulma-extensions/bulma-tooltip/src/sass/index.sass';
```
Comment (abhijeet-qure) [inline portal/client/src/scss/vendor.scss:8]: are you planning to use these components?

File: portal/client/src/App.js [hunk 23]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import './scss/index.scss'
+import Router from './router'
+
+function App() {
+  return (
+    <Router />
```
Comment (abhijeet-qure) [inline portal/client/src/App.js:7]: this seems redundant, lets use the `Router` directly into the `index.js` file

File: portal/client/.eslintrc.js (hunk 2 above)
Comment (kevalnagda) [reply portal/client/.eslintrc.js:44]: no. not sure how it was added.

File: portal/client/README.md (hunk 3 above)
Comment (kevalnagda) [reply portal/client/README.md:1]: Sure, will update this.

File: portal/client/public/qTrackLogo.svg (hunk 8 above)
Comment (kevalnagda) [reply portal/client/public/qTrackLogo.svg:1]: No, I've added it for the moment, till you create the qStudy logo.

File: portal/client/src/App.css (hunk 9 above)
Comment (kevalnagda) [reply portal/client/src/App.css:1]: Will check and update.

File: portal/client/package.json (hunk 6 above)
Comment (kevalnagda) [reply portal/client/package.json:40]: Can't add TODO comments in json

File: portal/client/src/pages/StudyList/index.js (hunk 16 above)
Comment (kevalnagda) [reply portal/client/src/pages/StudyList/index.js:49]: We'll remove this.

File: portal/client/src/scss/vendor.scss (hunk 22 above)
Comment (kevalnagda) [reply portal/client/src/scss/vendor.scss:8]: nope

**Pull request merged**